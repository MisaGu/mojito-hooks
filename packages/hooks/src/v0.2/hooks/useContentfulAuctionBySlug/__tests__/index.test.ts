import { act, renderHook } from '@testing-library/react-hooks';
import { TestWrapper } from '../../../components/test/wrapper/TestWrapper';
import {
  useContentfulAuctionBySlug,
  UseContentfulFullLotProps,
  UseContentfulFullLotReturn,
} from '../useContentfulAuctionBySlug';

describe('useContentfulAuctionBySlug()', () => {
  it('should be defined', () => {
    expect(useContentfulAuctionBySlug).toBeDefined();
  });

  it('returns the right result when needed', async () => {
    const { result, rerender, waitFor } = renderHook<
      [UseContentfulFullLotProps],
      UseContentfulFullLotReturn
    >(
      (args) => {
        return useContentfulAuctionBySlug(...args);
      },
      {
        initialProps: [{ mojitoID: '5f6a59f0-63bb-4628-b0ff-fcad26559a09' }],
        wrapper: TestWrapper as any,
      },
    );

    // Trigger request...
    await act(async () => rerender());

    // ...and wait for response:
    await waitFor(() => result.current.queryResult.isSuccess);

    // Check data matches snapshot:
    expect(result.current).toMatchSnapshot({
      queryResult: {
        dataUpdatedAt: expect.any(Number),
      },
    });
  });
});
