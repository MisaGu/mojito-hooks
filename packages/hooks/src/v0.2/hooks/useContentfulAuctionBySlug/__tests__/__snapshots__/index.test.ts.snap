// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useContentfulAuctionBySlug() returns the right result when needed 1`] = `
Object {
  "error": null,
  "isLoading": false,
  "lots": Object {
    "data": null,
    "description": null,
    "duration": null,
    "endDate": "2022-03-18T12:00:00.000Z",
    "name": "bid test 3",
    "saleId": "BID_TEST_3",
    "slug": "bid-test-3",
    "startDate": "2022-03-16T12:00:00.000Z",
    "subtitle": null,
    "title": "bid test 3",
    "videoId": null,
  },
  "queryResult": Object {
    "dataUpdatedAt": Any<Number>,
    "errorUpdateCount": 0,
    "errorUpdatedAt": 0,
    "failureCount": 0,
    "isError": false,
    "isFetched": true,
    "isFetchedAfterMount": true,
    "isFetching": false,
    "isIdle": false,
    "isLoadingError": false,
    "isPlaceholderData": false,
    "isPreviousData": false,
    "isRefetchError": false,
    "isRefetching": false,
    "isStale": false,
    "isSuccess": true,
    "remove": [Function],
    "status": "success",
  },
  "refetch": [Function],
}
`;
